libs/features/personalization/personalization.js

```diff
@@ -216,11 +216,67 @@ function normalizeKeys(obj) {
   }, {});
 }
 
+const getDivInTargetedCell = (tableCell) => {
+  tableCell.replaceChildren();
+  const div = document.createElement('div');
+  tableCell.appendChild(div);
+  return div;
+};
+
+const querySelector = (el, selector, all = false) => {
+  try {
+    if (all) {
+      return el.querySelectorAll(selector);
+    }
+    return el.querySelector(selector);
+  } catch (e) {
+    /* eslint-disable-next-line no-console */
+    console.log('Invalid selector: ', selector);
+    return null;
+  }
+};
+
+function getSelectedEl(rootEl, selector) {
+  if (!selector) return null;
+
+  let selectedEl = querySelector(rootEl, selector);
+  if (selectedEl) return selectedEl;
+
+  // TODO: do we support space delimeted selectors?
+  const [firstTerm, secondTerm] = selector.split(' ').filter((str) => str?.trim());
+  if (firstTerm.includes('-')) {
+    const [, num] = firstTerm.split('-');
```

Comment:
Or, if the colon approach doesn't sound appealing, maybe we could split by a regex: dash, number, end of string